addons/mail/models/mail_activity.py

```diff
@@ -682,16 +682,23 @@ def get_activity_data(self, res_model, domain, limit=None, offset=0, fetch_done=
         res_id_to_date_done = {}
         res_id_to_deadline = {}
         grouped_activities = defaultdict(dict)
+        based_on_my_activities = self.env.context.get("my_activities", False)
```

Comment:
Don't like context-based functional choices, as it is easily broken and people don't know the output is modified by some magical contextual key. Would add a parameter to ``get_activity_data``, either something limited like ``my_only=False`` (which would further filter on my activities only), either something more open like ``activity_domain=None`` where we could give ``activity_domain=[('user_id', '=', uid)]`` when calling it.

And it should probably be used when computing ``all_activities`` above, unless there are reason to search all activities then filter ?

Reply:
This would probably allow to update other code place, as everything seems context-based, which is easily broken. Could you check to try to have real parameters and/or domains to propagate ?